splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java

```diff
@@ -129,16 +129,26 @@ private void configureHandler() throws Exception {
             // Transform message to add splunk wrapper to the json
             .transform().simple("{\"source\": \"eventing\", \"sourcetype\": \"Insights event\", \"event\": ${body}}")
 
+            // Redirect depending on http or https (different default ports) so that it goes to the default splunk port
             // Send the message to Splunk's HEC as a splunk formattted event.
             // It sends token via Basic Preemptive Authentication.
             // POST method is being used, set up explicitly
             // (see https://camel.apache.org/components/latest/http-component.html#_which_http_method_will_be_used).
-            .toD(http("$simple{headers.metadata[url]}/services/collector/event")
-                .authenticationPreemptive(true)
-                .authMethod("Basic")
-                .httpMethod("POST")
-                .authUsername("x")
-                .authPassword("$simple{headers.extras[token]}"))
+            .choice()
+                .when().simple("${headers.metadata[url]}" startsWith "https")
+                    .toD(http("https://$simple{headers.metadata[url][hostname]}:8088/services/collector/event")
+                        .authenticationPreemptive(true)
+                        .authMethod("Basic")
+                        .httpMethod("POST")
+                        .authUsername("x")
+                        .authPassword("$simple{headers.extras[token]}"))
+                .otherwise()
+                    .toD(http("http://$simple{headers.metadata[url]}:8088/services/collector/event")
```

Comment:
Since `headers.metadata[url]` contains the prefix of `http://` wouldn't this duplicate the prefix? Would it need a substring?

Can the port be carried within `headers.metadata[url]`, for cases when it is on a different port or on Splunk Cloud on a loadbalancer?